File: x-baseproject/src/main/java/com/xia/baseproject/ui/dialog/loader/Loader.java
package com.xia.baseproject.ui.dialog.loader;

import android.app.Dialog;

import com.xia.baseproject.ui.dialog.BaseDialog;

import java.util.ArrayList;

/**
 * @author xia
 * @date 2018/8/21.
 */
public final class Loader {
    private static final ArrayList<Dialog> LOADERS = new ArrayList<>();

    public static void showLoading(final Dialog dialog) {
        if (dialog != null) {
            LOADERS.add(dialog);
            BaseDialog.safeShowDialog(dialog);
        }
    }

    public static void stopLoading() {
        for (Dialog dialog : LOADERS) {
            BaseDialog.safeCloseDialog(dialog);
        }
        LOADERS.clear();
    }
}

File: fly/src/main/java/com/xia/fly/ui/dialog/loader/Loader.kt
package com.xia.fly.ui.dialog.loader

import android.app.Dialog
import com.xia.fly.ui.dialog.FlySupportDialog
import java.util.*

/**
 * @author xia
 * @date 2018/8/21.
 */
object Loader {
    private val LOADERS = ArrayList<Dialog>()

    @JvmStatic
    fun showLoading(dialog: Dialog?) {
        if (dialog != null) {
            LOADERS.add(dialog)
            FlySupportDialog.safeShowDialog(dialog)
        }
    }

    @JvmStatic
    fun stopLoading() {
        for (dialog in LOADERS) {
            FlySupportDialog.safeCloseDialog(dialog)
        }
        LOADERS.clear()
    }
}
